lib/statsd/instrument/client.rb

```diff
@@ -151,6 +153,30 @@ def histogram(name, value, sample_rate: nil, tags: nil)
 
   # @!endgroup
 
+  # Measures the latency of the given block in milliseconds, and emits it as a metric.
+  #
+  # @param name (see #increment)
+  # @param sample_rate (see #increment)
+  # @param tags (see #increment)
+  # @param [Symbol] metric_type The metric type to use. If not specified, we will
+  #   use the preferred metric type of the implementation. The default is `:ms`.
+  #   Generally, you should not have to set this.
+  # @yield The latency (execution time) of the block
+  # @return The return value of the proivded block will be passed through.
+  def latency(name, sample_rate: nil, tags: nil, metric_type: nil, &block)
+    start = Process.clock_gettime(Process::CLOCK_MONOTONIC)
+    block.call
```

Comment:
use `yield` and remove `&block` from the signature. You'll save one `Proc` allocation.

`&block` is only worth it if you pass it to another method.